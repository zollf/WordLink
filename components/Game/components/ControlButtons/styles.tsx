import { StyleSheet } from "react-native";

import AppStyle from "../../../../styles";

const styles = StyleSheet.create({
  buttons: {
    position: "absolute",
    bottom: 0,
    display: "flex",
    flexDirection: "row",
  },
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    width: 150,
    height: 40,
    backgroundColor: AppStyle.primary,
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2},
    shadowOpacity: 0.1,
    borderRadius: 5,
    margin: 20,
    bottom: 50,
  },
  text: {
    fontSize: 16,
    fontFamily: AppStyle.font,
  }
});

export default styles;