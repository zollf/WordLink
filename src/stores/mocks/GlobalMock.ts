const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
