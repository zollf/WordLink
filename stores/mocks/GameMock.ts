const level: Level = {
  title: 'hello',
  difficulty: 'easy',
  game: {
    start: 'FORTY',
    end: 'FIFTY',
    path: 5,
  },
};

const GameMocks = {
  gameId: 0,
  difficulty: 'EASY',
  inGame: false,
  loaded: false,
  error: false,
  game: level.game,
  gameTitle: 'hello',

  currentWord: level.game.start,
  selectedLetter: '',
  stepNumber: 0,
  path: [level.game.start],
  visited: { FORTY: true },
  completed: false,

  loadGame: jest.fn(),
  startGame: jest.fn(),
  getNewLetters: jest.fn(() => ['A', 'B', 'C']),
  isWord: jest.fn(),
  setSelectedLetter: jest.fn(),
  changeLetter: jest.fn(),
  resetCurrentWord: jest.fn(),
  undo: jest.fn(),
  clearGame: jest.fn(),
};

export default GameMocks;
