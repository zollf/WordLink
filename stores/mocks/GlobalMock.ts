const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
