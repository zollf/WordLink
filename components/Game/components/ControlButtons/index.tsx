import React from 'react';
import { View } from 'react-native';
import { useStore } from 'app/hooks';
import Button from 'app/components/Button';

import styles from './styles';

const ControlButtons = () => {
  const { resetCurrentWord, undo } = useStore('gameStore');

  return (
    <View style={styles.buttons}>
      <View style={styles.button}>
        <Button data-test-id="reset" color="orange" onPress={() => resetCurrentWord()} text="RESET" />
      </View>
      <View style={styles.button}>
        <Button data-test-id="undo" color="orange" onPress={() => undo()} text="UNDO" />
      </View>
    </View>
  );
};

export default ControlButtons;
