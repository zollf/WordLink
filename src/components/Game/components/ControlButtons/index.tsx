import React from 'react';
import { View } from 'react-native';
import { useStore } from 'app/src/hooks';
import Button from 'app/src/components/Button';

import styles from './styles';

const ControlButtons = () => {
  const { resetCurrentWord, undo } = useStore('gameStore');

  return (
    <View style={styles.buttons}>
      <View style={styles.button}>
        <Button data-test-id="reset" color="orange" onPress={() => resetCurrentWord()} text="RESET" />
      </View>
      <View style={styles.button}>
        <Button data-test-id="undo" color="orange" onPress={() => undo()} text="UNDO" />
      </View>
    </View>
  );
};

export default ControlButtons;
