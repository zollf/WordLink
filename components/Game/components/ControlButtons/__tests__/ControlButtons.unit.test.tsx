import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from 'app/stores/mocks/GlobalMock';
import GameMock from 'app/stores/mocks/GameMock';

import ControlButtons from '../';

import renderer from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock} gameStore={GameMock}>
    <ControlButtons />
  </Provider>
);

describe('<ControlButtons />', () => {
  let wrapper: renderer.ReactTestRenderer;
  beforeEach(() => {
    wrapper = renderer.create(component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls undo', () => {
    wrapper.root.findByProps({ 'data-test-id': 'undo' }).props.onPress();
    expect(GameMock.undo).toBeCalled();
  });

  it('calls reset', () => {
    wrapper.root.findByProps({ 'data-test-id': 'reset' }).props.onPress();
    expect(GameMock.resetCurrentWord).toBeCalled();
  });
});
