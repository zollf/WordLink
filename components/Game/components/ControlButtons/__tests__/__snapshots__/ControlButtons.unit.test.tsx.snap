// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlButtons /> matches snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "display": "flex",
      "flexDirection": "row",
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 26.5,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4294956160,
                4294950464,
              ],
              "endPoint": undefined,
              "locations": Array [
                0.5,
                0.5,
              ],
              "startPoint": Array [
                -0.1,
                0.1,
              ],
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 106.5,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Raleway_700Bold",
                "fontSize": 33.5,
              }
            }
          >
            RESET
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 26.5,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4294956160,
                4294950464,
              ],
              "endPoint": undefined,
              "locations": Array [
                0.5,
                0.5,
              ],
              "startPoint": Array [
                -0.1,
                0.1,
              ],
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 106.5,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Raleway_700Bold",
                "fontSize": 33.5,
              }
            }
          >
            UNDO
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;
